ui/components/app/nfts-tab/nfts-tab.js

```diff
@@ -82,6 +82,12 @@ export default function NftsTab() {
     });
   }, [showNftBanner, trackEvent, currentNetwork, currentLocale]);
 
+  useEffect(() => {
```

Comment:
If I have 100 NFTs, is this going to fire 100 RPC requests every time I switch to the NFT tab?  That seems like too much to do on every page load.  Would be more ok if it was behind a refresh button or when clicking a single NFT.

Reply:
We have a refresh button that does this and we refresh ownership also when a user clicks on the NFT.
I agree that this call might not be needed, sometimes users do not see the refresh button if they have many NFTs and they have to scroll all the way down too see it,, or might not have the reflex to click on their NFTs when they see it on their main page but they know they do not own it anymore. This scenario happens mainly when users sell the NFTs on marketplaces as opposed to manually transferring them (which triggers the ownerhsip update) 
There might something that can be done on core instead ...

Reply:
I think if NFT autodetection is on, the NFT API can be the source of truth.  If we can trigger those fetches at the right times, whatever comes back is what you own.

But if NFT autodetection is off, we will need to do RPC requests to check if you're still the owner.  I'm thinking we could do a similar optimization to how erc20 balances are updated.  Meaning, a multicall style RPC request that can check ownership/balance for all NFTs in 1 RPC request.